bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/ephemeris/TcsEphemerisCron.scala

```diff
@@ -61,6 +71,70 @@ object TcsEphemerisCron {
     }
   }
 
+  private implicit class InstantOps(i: Instant) {
+    /** Returns this instant truncated to minutes (ie, with no seconds,
+      * milliseconds, etc.)
+      */
+    def roundDown: Instant =
+      i.truncatedTo(MINUTES)
+
+    /** Returns this instant rounded up to the next whole number of minutes
+      * with no seconds, milliseconds, etc.)
+      */
+    def roundUp: Instant = {
+      val rnd = i.roundDown
+      (rnd == i) ? rnd | rnd.plus(Duration.ofMinutes(1))
+    }
+  }
+
+  /** A Night implementation for a specific start and end time. */
+  private class SpecificNight(site: Site, s: Instant, e: Instant) extends Night {
+    val start = s.toEpochMilli
+    val end   = e.toEpochMilli
+
+    override def getSite: Site      = site
+    override def getStartTime: Long = start
+    override def getEndTime: Long   = end
+    override def getTotalTime: Long = end - start
+
+    override def includes(time: Long): Boolean =
+      (start <= time) && (time < end)
```

Comment:
I'm not sure I understand the question.  Here we're implementing an existing interface where times are expressed as epoch milliseconds.